File: WordCount.py
from pyspark import SparkContext, SparkConf
import sys

inputs = sys.argv[1]
outputs = sys.argv[2]

conf = SparkConf().setAppName('word count')
conf.set("spark.hadoop.validateOutputSpecs", "false")
context = SparkContext(conf=conf)
context._jsc.hadoopConfiguration().set("textinputformat.record.delimiter", " ")

rdd = context.textFile(inputs)				\
			 .filter(lambda x:x!='')		\
			 .map(lambda x:(x.lower(), 1))	\
			 .reduceByKey(lambda x,y:x+y)	\
			 .coalesce(1)					\
			 .saveAsTextFile(outputs)

File: Scala/WordCount/WordCount.scala
import org.apache.spark._
import org.apache.spark.SparkContext._

object WordCount	{
	def main(args: Array[String])	{
		val inputFile = args(0)
		val outputFile = args(1)
		val conf = new SparkConf().setAppName("word count")
		val sc = new SparkContext(conf)

		val rdd = sc.textFile(inputFile)
		val rddSplit = rdd.flatMap(line => line.split(" "))
		val rddWordTuple = rddSplit.flatMap(word => (word, 1))
		val result = rddWordTuple.reduceByKey((count1, count2) => count1 + count2)

		result.saveAsTextFile(outputFile)
	}
}